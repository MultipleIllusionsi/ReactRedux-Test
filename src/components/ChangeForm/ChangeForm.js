import React from "react";
import PropTypes from "prop-types";

import "./ChangeForm.scss";

const ChangeForm = ({ onSubmit, handlerTitle, handlerBody, text }) => (
  <form className="form-edit" onSubmit={onSubmit}>
    <div>
      <label htmlFor="title">Title</label>
      <input
        autoFocus
        onChange={handlerTitle}
        id="title"
        type="text"
        placeholder="Title..."
        required
      />
    </div>

    <div>
      <label htmlFor="text">Text</label>
      <textarea
        onChange={handlerBody}
        id="text"
        rows="10"
        placeholder="Text..."
        required
      ></textarea>
    </div>

    <button type="submit">{text}</button>
  </form>
);

ChangeForm.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  handlerTitle: PropTypes.func.isRequired,
  handlerBody: PropTypes.func.isRequired,
  text: PropTypes.string
};

export default ChangeForm;
