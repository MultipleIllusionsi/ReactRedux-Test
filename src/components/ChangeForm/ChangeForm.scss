.form-edit {
    margin-top: 5vh;
    border: 1px solid #a4b0be;
    padding: 3rem;
    box-shadow: 0 0.2rem 0.3rem 0 black;
    transition: 0.2s all ease-out;

    &:hover {
        box-shadow: 0 0.5rem 0.3rem 0.2rem black;
    }

    & div {
        display: flex;
        flex-direction: column;

        &:not(:last-child) {
            margin-bottom: 3rem;
        }

        & label {
            font-size: 3rem;
        }

        & input,
        & textarea {
            padding: 1rem;
            font-size: 2rem;
            font-weight: normal;
        }

        & textarea {
            resize: vertical;
            font-family: inherit;
        }
    }

    & button {
        background-color: #686de0;
        padding: 1rem 3rem;
        font-size: 2.5rem;
        color: white;
        display: block;
        margin: 0 auto;
        border: none;
        transition: 0.3s all ease-out;
        cursor: pointer;

        &:hover {
            background-color: #4834d4;
        }
    }
}