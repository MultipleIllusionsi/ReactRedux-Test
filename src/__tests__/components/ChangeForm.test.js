import { shallow } from "enzyme";
import React from "react";
import ChangeForm from "../../components/ChangeForm/ChangeForm";

it("expect to render ChangeForm component", () => {
  const component = shallow(
    <ChangeForm
      onSubmit={() => {}}
      handlerTitle={() => {}}
      handlerBody={() => {}}
      text="Save"
    />
  );
  expect(component.debug()).toMatchSnapshot();
});
