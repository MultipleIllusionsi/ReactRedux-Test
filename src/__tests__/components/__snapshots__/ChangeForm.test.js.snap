// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render ChangeForm component 1`] = `
"<form className=\\"form-edit\\" onSubmit={[Function: onSubmit]}>
  <div>
    <label htmlFor=\\"title\\">
      Title
    </label>
    <input autoFocus={true} onChange={[Function: handlerTitle]} id=\\"title\\" type=\\"text\\" placeholder=\\"Title...\\" required={true} />
  </div>
  <div>
    <label htmlFor=\\"text\\">
      Text
    </label>
    <textarea onChange={[Function: handlerBody]} id=\\"text\\" rows=\\"10\\" placeholder=\\"Text...\\" required={true} />
  </div>
  <button type=\\"submit\\">
    Save
  </button>
</form>"
`;
